openhands/resolver/send_pull_request.py

```diff
@@ -518,7 +530,39 @@ def process_single_issue(
 
     apply_patch(patched_repo_dir, resolver_output.git_patch)
 
-    make_commit(patched_repo_dir, resolver_output.issue, issue_type)
+    has_changes = make_commit(patched_repo_dir, resolver_output.issue, issue_type)
+
+    # If there are no changes, we still want to post a comment with the result explanation
+    if not has_changes:
+        # Create a handler to post comments for both issues and PRs
+        handler = (
+            ServiceContextIssue(
+                GithubIssueHandler(
+                    resolver_output.issue.owner,
+                    resolver_output.issue.repo,
+                    token,
+                    username,
+                    base_domain,
+                ),
+                llm_config,
+            )
+            if platform == ProviderType.GITHUB
+            else ServiceContextIssue(
+                GitlabIssueHandler(
+                    resolver_output.issue.owner,
+                    resolver_output.issue.repo,
+                    token,
+                    username,
+                    base_domain,
+                ),
+                llm_config,
+            )
+        )
```

Comment:
NIT: I think we have similar logic for `resolve_issue`, ideally we pull that out to a shared method

Not blocking though!